File: web/src/main/java/no/nav/fo/mia/config/ApplicationConfig.java
package no.nav.fo.mia.config;

import org.springframework.context.annotation.*;

@EnableAspectJAutoProxy
@Configuration
@Import({
        ContentConfig.class,
        EndpointsConfig.class,
        MetricsConfig.class,
        CacheConfig.class
})
@ComponentScan(basePackages = {"no.nav.fo.mia.rest", "no.nav.fo.mia.internal.ressurser"})
public class ApplicationConfig {

}

File: src/main/kotlin/no/nav/fo/mia/config/ApplicationConfig.kt
package no.nav.fo.mia.config

import no.nav.fo.mia.util.CorsFilter
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.boot.web.servlet.FilterRegistrationBean
import org.springframework.cache.annotation.EnableCaching


@Configuration
@EnableCaching
open class ApplicationConfig {
    @Bean
    open fun corsFilterRegistration(): FilterRegistrationBean<CorsFilter> {
        val frb: FilterRegistrationBean<CorsFilter> = FilterRegistrationBean()
        frb.filter = CorsFilter()
        frb.urlPatterns = arrayListOf("/*")
        return frb
    }
}
